import/index_java/src/org/vufind/index/WorkKeys.java

```diff
@@ -0,0 +1,109 @@
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+import org.marc4j.marc.Record;
+import org.marc4j.marc.DataField;
+import org.marc4j.marc.Subfield;
+
+import org.vufind.index.FieldSpecTools;
+
+import java.text.Normalizer;
+import java.text.Normalizer.Form;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Indexing routines for creating work keys for FRBR functionality.
+ */
+public class WorkKeys
+{
+    /**
+     * Get all work identification keys for the record.
+     *
+     * @param  record              MARC record
+     * @param  uniformTitleTagList The field specification for uniform titles
+     * @param  titleTagList        The field specification for titles
+     * @param  authorTagList       The field specification for authors
+     * @param  includeRegEx        Regular expression defining characters to keep
+     * @param  excludeRegEx        Regular expression defining characters to remove
+     * @return set of keys
+     */
+    public Set<String> getWorkKeys(final Record record, final String uniformTitleTagList,
+        final String titleTagList, final String authorTagList, final String includeRegEx,
+        final String excludeRegEx
+
+    ) {
+        Set<String> workKeys = new LinkedHashSet<String>();
+
+        // Uniform title
+        final Set<String> uniformTitles = FieldSpecTools.getFieldsByTagList(record, uniformTitleTagList);
+        for (String uniformTitle : uniformTitles) {
+            final String normalizedUniformTitle = normalizeWorkKey(uniformTitle, includeRegEx, excludeRegEx);
+            if (!normalizedUniformTitle.isEmpty()) {
+                workKeys.add("UT ".concat(normalizedUniformTitle));
+            }
+        }
+
+        // Title + Author
+        final Set<String> titles = FieldSpecTools.getFieldsByTagList(record, titleTagList);
+        final Set<String> authors = FieldSpecTools.getFieldsByTagList(record, authorTagList);
+
+        for (String title : titles) {
+            final String normalizedTitle = normalizeWorkKey(title, includeRegEx, excludeRegEx);
+            if (!normalizedTitle.isEmpty()) {
+                for (String author : authors) {
+                    final String normalizedAuthor = normalizeWorkKey(author, includeRegEx, excludeRegEx);
+                    if (!normalizedAuthor.isEmpty()) {
+                        workKeys.add("AT ".concat(normalizedAuthor).concat(" ").concat(normalizedTitle));
+                    }
+                }
+            }
+        }
+
+        return workKeys;
+    }
+
+    /**
+     * Create a key string
+     *
+     * @param  s      String to normalize
+     * @param  regExp Regular expression defining characters to strip out
```

Comment:
Just happened to notice that the documentation doesn't quite match the actual argument list here.

Reply:
Should be fixed now.